### PR TITLE
Add CGM diabetes classification as stage 6 of curriculum learning.

### PR DESCRIPTION
**Pull request by tiageri**
# *Add CGM diabetes classification as curriculum learning stage 6*

## :recycle: Current situation & Problem
*This PR extends the curriculum learning framework with a sixth stage for continuous glucose monitor (CGM) data, enabling the model to classify diabetes status (healthy, pre diabetes, oral medication, insulin-dependent) from multi-day CGM recordings with chain of thought rationales.

## :gear: Release Notes
New cgm_diabetes package (cgm_diabetes/data/, cgm_diabetes/captioning/):
- cgm_loader.py — downloads and caches patient CGM data from Azure Blob Storage as local parquet files
- participants.py — loads the participant manifest from Azure, filtering to patients with CGM data and one of four diabetes labels
- CGMDiabetesDataset.py — torch.utils.data.Dataset compatible with the OpenTSLM pipeline; returns normalized glucose time series, summary statistics, and GPT-4o chain-of-thought answers (or template rationales if no GPT-4o generated captions available)
- generate_captions.py — generates GPT-4o vision captions from CGM plots and statistics; supports dry run, incremental saves, and --force_regenerate
- curriculum_learning.py: adds stage6_cgm_cot with the same LoRA + CoT setup as stages 4–5, and a cgm_accuracy metric function
- requirements.txt adds azure-storage-blob and python-dotenv

Note: Stage 6 requires access to a private CGM dataset on Azure Blob Storage. .env.example documents required Azure and OpenAI credentials.

## :books: Documentation
- README.md updated with stage 6 documentation
- .env.example documents required environment variables
- In-line commends on all functions and dataset class

## :white_check_mark: Testing
*Please ensure that the PR meets the testing requirements set by CodeCov and that new functionality is appropriately tested.*
*This section describes important information about the tests and why some elements might not be testable.*
Stage 6 depends on a private Azure-hosted dataset and is not automatically testable without credentials. Manual testing performed on caption generation (sample captions generated for five patients, compared against CGM plots). CGMDiabetesDataset verified to be compatible with existing training loop.

### Code of Conduct & Contributing Guidelines
By creating and submitting this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

## Release Notes

* **New Features**
  * Added Stage 6: CGM Diabetes Classification, a new training curriculum stage
  * Integrated GPT-4o vision-based caption generation for continuous glucose monitoring data
  * Added Azure Blob Storage integration for secure CGM data loading and caching
  * Introduced new dataset support with metadata filtering and automatic data prefetching

* **Documentation**
  * Added configuration guide and Stage 6 setup instructions in README
  * Created `.env.example` template for required environment variables

* **Chores**
  * Updated project dependencies for CGM data processing and AI vision capabilities

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copy this file to .env and fill in values.
+
+# Azure Blob Storage — required for CGM data (stage 6)
+AZURE_ACCOUNT_NAME=your_storage_account_name
+AZURE_SAS_TOKEN=your_sas_token
+AZURE_CONTAINER_NAME=your_container_name
+AZURE_DATASET_PREFIX=your_dataset_prefix
```
[the remark this answers is not in the document]
Comment (tiageri) [reply .env.example:7]: Will fix in followup.

File: cgm_diabetes/data/prefetch.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from cgm_diabetes.data.CGMDiabetesDataset import CGMDiabetesDataset
+import sys
+
+split = sys.argv[1] if len(sys.argv) > 1 else None
+CGMDiabetesDataset.prefetch(split)
```
[the remark this answers is not in the document]
Comment (tiageri) [reply cgm_diabetes/data/prefetch.py:5]: Will fix in followup.

File: curriculum_learning.py [hunk 3]
```diff
@@ -1417,6 +1428,39 @@ def stage5_ecg_cot(
...
+        - Metric: Test loss only
+        """
+
+        def cgm_accuracy(predictions, gold_answers):
+            correct = sum(
+                1 for pred, gold in zip(predictions, gold_answers)
+                if gold.split("Answer:")[-1].strip() ==
+                pred.split("Answer:")[-1].strip().split()[0].rstrip(".,;:")
+            )
+            return {"accuracy": correct / len(predictions) if predictions else 0.0}
```
[the remark this answers is not in the document]
Comment (tiageri) [reply curriculum_learning.py:1388]: Will fix in followup.
